src/components/Accordion/Accordion.tsx

```diff
@@ -126,18 +151,33 @@ export const Accordion: FC<AccordionProps> = (props) => {
     const {
         // @react-aria prevents default action for onMouseDown as implemented here: https://github.com/adobe/react-spectrum/blob/e14523fedd93ac1a4ede355aed70988af572ae74/packages/%40react-aria/selection/src/useSelectableCollection.ts#L370
         // This makes it impossible to edit or select text in input fields inside the accordion
+        // onKeydown automatically navigates away from child elements, including dropdown menus etc.
+        // Focus should only be passed if the accordian item itself is focused
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        accordionProps: { onMouseDown, ...accordionProps },
+        accordionProps: { onMouseDown, onKeyDown, ...accordionProps },
         // @react-aria enable by default typeahead which result in an event fired up on keypress and select the section
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         //@ts-ignore
     } = useAccordion({ ...ariaProps, disallowTypeAhead: true }, state, ref);
 
+    const accordionItemCheck = (event: KeyboardEvent<HTMLDivElement>) => {
+        const validItemIds = [ACCORDION_ID, ACCORDION_ITEM_ID];
+        const id = (event.target as HTMLButtonElement).dataset?.testId;
```

Comment:
`data-test-id` are only available in development/testing environment and removed in the production build, so we shouldn't rely on it.

Reply:
you could probably check if the actual item is focused in the selection manager from the item state

Reply:
Fixed :)